examples/scaling/main.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"image/png"
+	"log"
+	"os"
+
+	"github.com/asticode/go-astiav"
+)
+
+func main() {
+	var (
+		dstFilename string
+		dstWidth    int
+		dstHeight   int
+	)
+
+	flag.StringVar(&dstFilename, "output", "", "Output file name")
+	flag.IntVar(&dstWidth, "w", 0, "Destination width")
+	flag.IntVar(&dstHeight, "h", 0, "Destination height")
+	flag.Parse()
+
+	if dstFilename == "" || dstWidth <= 0 || dstHeight <= 0 {
+		fmt.Fprintf(os.Stderr, "Usage: %s -output output_file -w W -h H\n", os.Args[0])
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	dstFile, err := os.Create(dstFilename)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not open destination file %s\n", dstFilename)
+		os.Exit(1)
+	}
+	defer dstFile.Close()
+
+	srcW, srcH := 320, 240
+	srcPixFmt, dstPixFmt := astiav.PixelFormatYuv420P, astiav.PixelFormatRgba
+	srcFrame := astiav.AllocFrame()
+	srcFrame.SetHeight(srcH)
+	srcFrame.SetWidth(srcW)
+	srcFrame.SetPixelFormat(srcPixFmt)
+	srcFrame.AllocBuffer(1)
+	srcFrame.ImageFillBlack()
```

Comment:
You need to check error here. Message should be `fmt.Errorf("main: filling source frame with black image failed: %w", err)`